<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-03-12" />

<title>Lister_lab</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lister_lab_FLAMESv2_L1775</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Lister_lab</h1>
<h4 class="date">2025-03-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>lister_lab_FLAMESv2_L1775/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250312code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250312)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250312code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250312)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-8
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>about_cache</code> and re-run the analysis. To have workflowr
automatically delete the cache directory prior to building the file, set
<code>delete_cache = TRUE</code> when running <code>wflow_build()</code>
or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/
</td>
<td style="text-align:left;">
output
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/geneSymbol_gene_count.csv
</td>
<td style="text-align:left;">
output/geneSymbol_gene_count.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/Lister_L1775_with_doublets_umap_object.rds
</td>
<td style="text-align:left;">
output/Lister_L1775_with_doublets_umap_object.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/Lister_L1775_umap_object.rds
</td>
<td style="text-align:left;">
output/Lister_L1775_umap_object.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/gene_symbol_oarfish_counts.csv
</td>
<td style="text-align:left;">
output/gene_symbol_oarfish_counts.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/Lister_L1775_gene_and_isoform_seurat.rds
</td>
<td style="text-align:left;">
output/Lister_L1775_gene_and_isoform_seurat.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong3b2b47b">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 3b2b47b
</a>
</p>
</div>
<div id="strongRepositoryversionstrong3b2b47b"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 3b2b47b.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/about_cache/

Untracked files:
    Untracked:  output/Lister_L1775_gene_and_isoform_seurat.rds
    Untracked:  output/Lister_L1775_stats.csv
    Untracked:  output/Lister_L1775_umap_object.rds
    Untracked:  output/Lister_L1775_with_doublets_umap_object.rds
    Untracked:  output/geneSymbol_gene_count.csv
    Untracked:  output/gene_symbol_oarfish_counts.csv
    Untracked:  output/isoform_gene_dict.csv
    Untracked:  outputLister_L1775_gene_and_isoform_seurat.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/about.Rmd</code>) and HTML
(<code>docs/about.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
3b2b47b
</td>
<td>
Sefi Prawer
</td>
<td>
2025-03-12
</td>
<td>
Lister_lab first commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
89531b1
</td>
<td>
Sefi Prawer
</td>
<td>
2025-03-12
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="lister-lab-analysis-flamesv2" class="section level1">
<h1>Lister Lab analysis FLAMESv2</h1>
<div id="preamble" class="section level2">
<h2>Preamble</h2>
<p>Sample: L1775<br />
FALMESv2 run completed on 11 March 2025<br />
Bambu Version: 3.3.5<br />
NDR: 0.048<br />
<br />
<u>I had to do a lot of troubleshooting to get FLAMES to run. This
included:</u></p>
<ul>
<li><p>Filtering the genome alignment to remove supplementary
alignments.</p></li>
<li><p>Using HongYang’s fix for the Bambu version; otherwise, it throws
an error. This fix can be found in the SLURM output files in the
<code>outs</code> folder.</p></li>
<li><p>Manually adding BC and UMI tags to the transcriptome alignment
file using the old matched <code>reads.fastq</code> file.</p></li>
<li><p>Sorting the file by read name.</p></li>
</ul>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code>library(rtracklayer)
library(Seurat)
library(DropletUtils)
library(gridExtra)
library(BiocParallel)
library(SingleCellExperiment)
library(DoubletFinder)
library(stringr)
library(cowplot)
library(grid)
library(patchwork)
library(tidyverse)
library(Matrix)
library(data.table)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>define fucntions for gene dict</p>
<pre class="r"><code># Function to make csv naming resource 
make_isoform_gene_symbol_dict &lt;- function(FLAMES_gtf, 
                                          reference_gtf, 
                                          output_file) {
  # Import the first GTF file (transcripts GTF)
  gtf1 &lt;- import(FLAMES_gtf)
  gtf1_df &lt;- as.data.frame(gtf1)
  
  # Select relevant columns from the first GTF
  selected_columns1 &lt;- gtf1_df[, c(&quot;transcript_id&quot;, &quot;gene_id&quot;)]
  unique_selected_cols &lt;- unique(selected_columns1)
  
  # Import the second GTF file (reference GTF with gene symbols)
  gtf2 &lt;- import(reference_gtf)
  gtf2_df &lt;- as.data.frame(gtf2)
  
  # Select relevant columns from the second GTF
  selected_columns2 &lt;- gtf2_df[, c(&quot;gene_name&quot;, &quot;gene_id&quot;)]
  unique_gene_symbol &lt;- unique(selected_columns2)
  
  # Merge the two data frames on &#39;gene_id&#39;
  combined_data &lt;- merge(unique_selected_cols, 
                         unique_gene_symbol, 
                         by = &quot;gene_id&quot;, 
                         all.x = TRUE)
  
  # If &#39;gene_name&#39; is missing, replace it with &#39;gene_id&#39;
  combined_data$gene_symbol &lt;- ifelse(is.na(combined_data$gene_name), 
                                      combined_data$gene_id, 
                                      combined_data$gene_name)
  
  # Select relevant columns
  final_combined_data &lt;- combined_data[, c(&quot;transcript_id&quot;, &quot;gene_id&quot;, &quot;gene_symbol&quot;)]
  
  # Write to a CSV file
    write.csv(final_combined_data, file = file.path(&quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/&quot;, output_file), row.names = FALSE)

  
  return(final_combined_data)
}

#Convert count matrices from Gene ID to gene Symbol

# The FLAMES ref can be found in your selected output folder after running the Flames pipeline. 
FLAMES_gtf_file &lt;- &quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/outs/isoform_annotated.gtf&quot; #ensure file is unzipped
reference_gtf_file &lt;- &quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/resources/gencode.v47.annotation.gtf&quot; # ensure file is unzipped
output_file &lt;- &quot;isoform_gene_dict.csv&quot;

# Call the helper function defined in code block above to create a dictionary containing corresponding gene information for each isoform
# This may take a few minutes 
isoform_gene_dict &lt;- make_isoform_gene_symbol_dict(FLAMES_gtf_file,
                                                   reference_gtf_file,
                                                   output_file)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div id="convert-count-matrices-from-gene-id-to-gene-symbol"
class="section level2">
<h2>Convert count matrices from Gene ID to gene Symbol</h2>
<pre class="r"><code>convert_ENSGID_to_geneSymbol &lt;- function(gene_count_matrix_path, 
                                         id_symbol_df = isoform_gene_dict, 
                                         output_file,
                                         return_df = TRUE) {
  
  # Load the reference dictionary we made earlier - select gene-level cols
  id_symbol_df &lt;- as_tibble(id_symbol_df) %&gt;%
    dplyr::select(gene_id, gene_symbol)
  
  # Load the data object with ENSGID row names
  gene_count_matrix &lt;- fread(gene_count_matrix_path, header = TRUE)
  colnames(gene_count_matrix)[1] &lt;- &quot;gene_id&quot;
  
  # Replace ENSGIDs with gene symbols in original flames gene-level count matrix
  formatted_gene_count_matrix &lt;- gene_count_matrix %&gt;%
  merge(id_symbol_df, by.x = &#39;gene_id&#39;, by.y = &#39;gene_id&#39;) %&gt;%   # Add gene symbol information
    distinct(gene_symbol, .keep_all = TRUE) %&gt;%   # Remove duplicates based on gene symbol
    dplyr::select(-gene_id) %&gt;%   # Remove the ENSGID column
    column_to_rownames(var = &quot;gene_symbol&quot;)   # use the gene symbols we added as rownames
  
  # Write out the processed data frame
  fwrite(formatted_gene_count_matrix, 
            output_file, 
            row.names = TRUE)
  
  # Return the processed count matrix for further use if needed
  if(return_df){
    return(formatted_gene_count_matrix)
  }
}


# convert Gene_id to gene symbol for gene counts
gene_counts &lt;- convert_ENSGID_to_geneSymbol(
  gene_count_matrix_path = &quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/outs/gene_count.csv&quot;,
  output_file = &quot;output/geneSymbol_gene_count.csv&quot;
)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Execute the above code and save intermediate files.<br />
Here i set many code chunck to eval=FALSE, to limit size of workflow
docs.</p>
<pre class="r"><code># standard QC filtering and also remove doublets 
#define sample name
sample_id = &#39;Lister_L1775&#39;

# Read the CSV file
gene_counts &lt;- fread(&quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/geneSymbol_gene_count.csv&quot;)

# Convert to data frame to support row names
gene_counts &lt;- as.data.frame(gene_counts)

# Set the first column (V1) as row names and remove it from the data table
rownames(gene_counts) &lt;- gene_counts$V1

gene_counts$V1 &lt;- NULL

# Create Seurat object
seurat_object &lt;- CreateSeuratObject(counts = gene_counts, project = sample_id)</code></pre>
<pre><code>Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre><code>Warning: Data is of class data.frame. Coercing to dgCMatrix.</code></pre>
<pre class="r"><code># Plot relationship between reads and unique genes per cell
plot_scatter1 &lt;- FeatureScatter(seurat_object, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;) +
  geom_smooth(method = &quot;lm&quot;) + NoLegend() +
  labs(title = &quot;Reads vs Unique Genes per Cell BEFORE Filtering&quot;)

plot(plot_scatter1)</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/about.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="perform-very-basic-qc" class="section level2">
<h2>Perform very basic QC</h2>
<p>Will perform very basic QC. This should probably be done in
conjunction with data derived from Chucks paper in Cell. The
distributions of cell counts here look more complex.</p>
<pre class="r"><code># Add mitochondrial percentage
seurat_object[[&quot;joined&quot;]] &lt;- JoinLayers(seurat_object[[&quot;RNA&quot;]])</code></pre>
<pre><code>Warning: Key &#39;rna_&#39; taken, using &#39;joined_&#39; instead</code></pre>
<pre class="r"><code>seurat_object[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(seurat_object, pattern = &quot;^MT-&quot;)

p1 &lt;- VlnPlot(seurat_object, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;))</code></pre>
<pre><code>Warning: Default search for &quot;data&quot; layer in &quot;RNA&quot; assay yielded no results;
utilizing &quot;counts&quot; layer instead.</code></pre>
<pre class="r"><code>p1 +  plot_annotation(title = &quot;QC plots (gene level) BEFORE Filtering&quot;)</code></pre>
<p><img src="figure/about.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Filter cells based on feature and count thresholds
## define the filtering params - (change these based on your data)
max.features = 10000
min.features = 1000
min.counts = 1000
max.counts = 100000
MT = 10
npc = 15
doublet_rate = 0.08 ### I have no idea what the doublet rate is here will need to chek with chuck will assume 10,000 cells target
cluster_res = 0.9


#now we filter the seurat object based on the QC params listed above 
filt_seurat_object &lt;- subset(seurat_object, subset = nFeature_RNA &gt; min.features &amp; nFeature_RNA &lt; max.features &amp; percent.mt &lt; MT &amp; nCount_RNA &lt; max.counts &amp; nCount_RNA &gt; min.counts)

# Plot quality metrics after filtering
p2 &lt;- VlnPlot(filt_seurat_object, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;))</code></pre>
<pre><code>Warning: Default search for &quot;data&quot; layer in &quot;RNA&quot; assay yielded no results;
utilizing &quot;counts&quot; layer instead.</code></pre>
<pre class="r"><code>p2 + plot_annotation(title = &quot;QC metrics gene level AFTER Filtering&quot;)</code></pre>
<p><img src="figure/about.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dim(filt_seurat_object)</code></pre>
<pre><code>[1] 39906 10660</code></pre>
<pre class="r"><code>dim(seurat_object)</code></pre>
<pre><code>[1] 39906 10794</code></pre>
<pre class="r"><code># Normalize data
filt_seurat_object &lt;- NormalizeData(filt_seurat_object, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)
  
# Identify highly variable features
filt_seurat_object &lt;- FindVariableFeatures(filt_seurat_object, selection.method = &quot;vst&quot;, nfeatures = 2000)
  
# Apply linear transformation
all_genes &lt;- rownames(filt_seurat_object)
filt_seurat_object &lt;- ScaleData(filt_seurat_object, features = all_genes)
  
# Perform PCA
filt_seurat_object &lt;- RunPCA(filt_seurat_object, features = VariableFeatures(object = filt_seurat_object))
  
# Cluster cells
filt_seurat_object &lt;- FindNeighbors(filt_seurat_object, dims = 1:npc)
filt_seurat_object &lt;- FindClusters(filt_seurat_object, resolution = cluster_res)

# Perform UMAP
filt_seurat_object &lt;- RunUMAP(filt_seurat_object, dims = 1:npc)</code></pre>
</div>
</div>
<div id="finding-and-removing-doublets" class="section level1">
<h1>Finding and removing doublets</h1>
<pre class="r"><code>### Filter out doublets (remember to modify doublet rate if samples have variable target cells)
## pK Identification (no ground-truth) 
sweep.res.list_pbmc &lt;- paramSweep(filt_seurat_object, PCs = 1:npc, sct = FALSE)


sweep.stats_pbmc &lt;- summarizeSweep(sweep.res.list_pbmc, GT = FALSE)
bcmvn_pbmc &lt;- find.pK(sweep.stats_pbmc)

#####
pK &lt;- bcmvn_pbmc %&gt;% filter(BCmetric == max(BCmetric)) %&gt;% dplyr::select(pK) 
pK &lt;- as.numeric(as.character(pK[[1]]))
  
## Homotypic Doublet Proportion Estimate 
annotations &lt;- filt_seurat_object@meta.data$seurat_clusters
homotypic.prop &lt;- modelHomotypic(annotations)
nExp_poi &lt;- round(doublet_rate * nrow(filt_seurat_object@meta.data))
nExp_poi.adj &lt;- round(nExp_poi * (1 - homotypic.prop))
  
  # Run doubletFinder 
filt_seurat_object &lt;- doubletFinder(filt_seurat_object, PCs = 1:npc, pN = 0.25, pK = pK, nExp = nExp_poi.adj, reuse.pANN = FALSE, sct = FALSE)

colnames(filt_seurat_object@meta.data) &lt;- sub(&quot;DF.classifications_.*$&quot;, &quot;DF.classifications&quot;, colnames(filt_seurat_object@meta.data))

  
  
###Save the seurat object with doublets listed 
filt_seurat_object_doublets &lt;- filt_seurat_object
filt_seurat_object &lt;- subset(filt_seurat_object, subset = DF.classifications == &#39;Singlet&#39;)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<div id="doublet-plots" class="section level2">
<h2>doublet plots</h2>
<pre class="r"><code>#load in seurats from previous step
filt_seurat_object_doublets &lt;- readRDS(&quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/Lister_L1775_with_doublets_umap_object.rds&quot;)

filt_seurat_object &lt;- readRDS(&quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/Lister_L1775_umap_object.rds&quot;)

# figures
  ggplot_list &lt;- list(
  ElbowPlot(filt_seurat_object) + labs(title = &#39;SD explained by each PC&#39;) + theme(text = element_text(size = 10)),
  
  FeatureScatter(filt_seurat_object, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;) +
    geom_smooth(method = &quot;lm&quot;) + NoLegend() + labs(title = &quot;Association between reads and \nunique genes per cell AFTER filtering&quot;),
  
  DimPlot(filt_seurat_object, reduction = &quot;umap&quot;) + 
    labs(color = &quot;Cluster \n(from PCA)&quot;, title = &#39;&#39;) + 
    theme(text = element_text(size = 10)),
  
  FeaturePlot(filt_seurat_object, reduction = &quot;umap&quot;, features = &#39;nCount_RNA&#39;) + 
    labs(color = &quot;UMI count&quot;, title = &#39;&#39;) + 
    theme(text = element_text(size = 10)),
  
  FeaturePlot(filt_seurat_object, reduction = &quot;umap&quot;, features = &#39;nFeature_RNA&#39;) + 
    labs(color = str_wrap(&quot;Feature count (gene)&quot;, 15), title = &#39;&#39;) + 
    theme(text = element_text(size = 10)),
  
  p2
)
  

combined_plots &lt;- plot_grid(plotlist = ggplot_list, ncol = 3)</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<pre class="r"><code>  plot(combined_plots)</code></pre>
<p><img src="figure/about.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(DimPlot(filt_seurat_object_doublets, reduction = &#39;umap&#39;, group.by = &quot;DF.classifications&quot;))

# Summary doublets
statsDoublets &lt;- filt_seurat_object_doublets@meta.data %&gt;%
  group_by(DF.classifications) %&gt;%
  summarize(Median_nCount_RNA = median(nCount_RNA), Median_nFeature_RNA = median(nFeature_RNA), Count = n())


tbl_sts1 &lt;- tableGrob(statsDoublets)
  grid.draw(tbl_sts1)</code></pre>
<p><img src="figure/about.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="summary-stats" class="section level2">
<h2>Summary Stats</h2>
<pre class="r"><code>  stats_sumary &lt;- rbind(&quot;Sample ID&quot; = sample_id,
                        &quot;Cells_before_filter&quot; = dim(seurat_object)[2],
                        &quot;Cells_after_filter&quot; = dim(filt_seurat_object)[2],
                        &quot;Median Feature per Cell before filter&quot; = median(seurat_object$nFeature_RNA),
                        &quot;Median Reads per Gene before filter&quot; = median(seurat_object$nCount_RNA),
                        &quot;Median Feature per Cell&quot; = median(filt_seurat_object$nFeature_RNA),
                        &quot;Median Reads per Gene&quot; = median(filt_seurat_object$nCount_RNA),
                        &quot;Max Features&quot; = max.features,
                        &quot;Min Features&quot; = min.features,
                        &quot;Min Counts&quot; = min.counts,
                        &quot;Max Counts&quot; = max.counts,
                        &quot;MT Percentage&quot; = MT,
                        &quot;NPCs&quot; = npc,
                        &quot;Median Percent MT before Filter&quot; = median(seurat_object@meta.data[[&quot;percent.mt&quot;]]),
                        &quot;Median Percent MT after Filter&quot; = median(filt_seurat_object@meta.data[[&quot;percent.mt&quot;]]))
  
  tbl_sts2 &lt;- tableGrob(stats_sumary)
  grid.draw(tbl_sts2)</code></pre>
<p><img src="figure/about.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  #save files
  saveRDS(filt_seurat_object, file = paste0(&quot;output/&quot;, sample_id, &quot;_umap_object.rds&quot;))
  saveRDS(filt_seurat_object_doublets, file = paste0(&quot;output/&quot;,sample_id, &quot;_with_doublets_umap_object.rds&quot;))
  write.table(stats_sumary, file = paste0(&quot;output/&quot;, sample_id, &quot;_stats.csv&quot;)) </code></pre>
</div>
</div>
<div id="add-in-isoform-counts" class="section level1">
<h1>Add in isoform Counts</h1>
<pre class="r"><code>#This function reads in Oarfish count files and creates a csv file of count data. The function also appends the gene symbol to the ENSTID

process_oarfish_files_to_counts_matrix &lt;- function(flames_output_folder, sample_name, output_dir) {

  # Read in the resource table (transcript_id, gene_id, gene_symbol)
  # Define the file paths based on the sample name
  count_matrix_path &lt;- file.path(flames_output_folder, paste0(sample_name, &quot;.count.mtx&quot;))
  barcodes_path &lt;- file.path(flames_output_folder, paste0(sample_name, &quot;.barcodes.txt&quot;))
  features_path &lt;- file.path(flames_output_folder, paste0(sample_name, &quot;.features.txt&quot;))
  
  # Read the data
  counts &lt;- readMM(count_matrix_path)
  barcodes &lt;- readLines(barcodes_path)
  features &lt;- read.delim(features_path, header = FALSE)
  
  # Transpose the matrix for Seurat compatibility
  counts &lt;- t(counts)
  
  # Set row and column names
  rownames(counts) &lt;- features$V1
  colnames(counts) &lt;- barcodes
  
  # Convert to a data frame
  counts_df &lt;- as.data.frame(counts)
  
  # Add transcript_id as the first column
  counts_df$transcript_id &lt;- rownames(counts_df)
  counts_df &lt;- counts_df[, c(ncol(counts_df), 1:(ncol(counts_df)-1))]
  
  # Merge with the resource table to add gene symbols
  df_genesymbol &lt;- counts_df %&gt;%
    left_join(isoform_gene_dict, by = &quot;transcript_id&quot;)
  
  # Remove the gene_id column and reorder the columns
  df_genesymbol$gene_id &lt;- NULL
  df_genesymbol &lt;- df_genesymbol[, c(ncol(df_genesymbol), 1:(ncol(df_genesymbol)-1))]
  
  # Update row names to include gene symbol instead of transcript_id
  rownames(df_genesymbol) &lt;- paste0(df_genesymbol$transcript_id, &quot;_&quot;, df_genesymbol$gene_symbol)
  df_genesymbol$transcript_id &lt;- NULL
  df_genesymbol$gene_symbol &lt;- NULL
  
  # Write the output to a CSV file
  output_path &lt;- file.path(output_dir, paste0(&quot;gene_symbol_&quot;, sample_name, &quot;_counts.csv&quot;))
  # Write the df_genesymbol data frame to the CSV file using fwrite
  fwrite(df_genesymbol, output_path)
  
  cat(&quot;Processed sample:&quot;, sample_name, &quot;\nOutput saved to:&quot;, output_path, &quot;\n&quot;)
  
  return(df_genesymbol)
}

oarfish_counts &lt;- process_oarfish_files_to_counts_matrix(
  flames_output_folder = &quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/outs&quot;,
  sample_name = &quot;oarfish&quot;,
  output_dir = &quot;output/&quot;
    )</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p><br />
make a seurat object with oarfish counts and make assay. Add them back
to gene object</p>
<pre class="r"><code>  # Load the data object with ENSGID row names
# Read the CSV file
oarfish_counts &lt;- fread(&quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/gene_symbol_oarfish_counts.csv&quot;)

# Convert to data frame to support row names
oarfish_counts &lt;- as.data.frame(oarfish_counts)

# Set the first column (V1) as row names and remove it from the data table
rownames(oarfish_counts) &lt;- oarfish_counts$V1

oarfish_counts$V1 &lt;- NULL  
  
# Create the Seurat object with iso counts
iso_seurat_obj &lt;- CreateSeuratObject(counts = oarfish_counts, project = sample_id)

VlnPlot(iso_seurat_obj, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;), ncol = 2) + plot_annotation(title = &quot;QC plots (isoform level) BEFORE Filtering&quot;)

#### isofrom assay 
## filter the data so iso and gene cells match
oarfish_iso_matched_gene &lt;- subset(iso_seurat_obj, cells =filt_seurat_object@graphs[[&quot;RNA_nn&quot;]]@Dimnames[[1]])

#Rejoin data sets after integration
oarfish_iso_matched_gene &lt;- JoinLayers(oarfish_iso_matched_gene)
counts_table_iso &lt;- oarfish_iso_matched_gene[[&quot;RNA&quot;]]$counts

#as.data.frame(counts_table_iso) -&gt; df_iso

# Remove rows where the sum is 0
#df_iso &lt;- df_iso[rowSums(df_iso) != 0, ] # this kills the env will skip for now

#filt_seurat_object[[&quot;iso&quot;]] &lt;- CreateAssay5Object(counts = df_iso)
filt_seurat_object[[&quot;iso&quot;]] &lt;- CreateAssay5Object(counts = counts_table_iso)

# Normalize the new assay data
filt_seurat_object &lt;- NormalizeData(filt_seurat_object, assay = &quot;iso&quot;)
filt_seurat_object &lt;- FindVariableFeatures(filt_seurat_object, assay = &quot;iso&quot;)
filt_seurat_object &lt;- ScaleData(filt_seurat_object, assay = &quot;iso&quot;)
filt_seurat_object &lt;- RunPCA(filt_seurat_object, assay = &quot;iso&quot;, reduction.name = &quot;pca_iso&quot;)

#Run UMAP
filt_seurat_object &lt;- RunUMAP(filt_seurat_object, reduction = &quot;pca_iso&quot;, dims = 1:15,
                             assay = &quot;iso&quot;, reduction.name = &quot;umap_iso&quot;)

#check to see that we have two assays
filt_seurat_object</code></pre>
<p>save files to output</p>
<pre class="r"><code>saveRDS(filt_seurat_object, file = paste0(&quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/&quot;, sample_id, &quot;_gene_and_isoform_seurat.rds&quot;))</code></pre>
<div id="summary-table-for-mikes-grant" class="section level3">
<h3>Summary table for Mike’s grant</h3>
<pre class="r"><code>filt_seurat_object &lt;- readRDS(&quot;/data/gpfs/projects/punim2510/Project_listerlab/FLAMESv2/analysis/Lister_lab_FLAMESv2_L1775/lister_lab_FLAMESv2_L1775/output/Lister_L1775_gene_and_isoform_seurat.rds&quot;)

# visualize the UMAP for Gene and isoform
DimPlot(filt_seurat_object, label = TRUE, reduction = &quot;umap&quot;) + ggtitle(&quot;UMAP gene level clustering&quot;) |
DimPlot(filt_seurat_object, label = TRUE, reduction = &quot;umap_iso&quot;) + ggtitle(&quot;UMAP isoform level clustering&quot;)</code></pre>
<p><img src="figure/about.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(&quot;Isofrom features per cell&quot;)</code></pre>
<pre><code>[1] &quot;Isofrom features per cell&quot;</code></pre>
<pre class="r"><code>summary(filt_seurat_object$nFeature_iso)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    732    2132    3002    3643    5004   11383 </code></pre>
<pre class="r"><code>print(&quot;Isofrom counts per cell&quot;)</code></pre>
<pre><code>[1] &quot;Isofrom counts per cell&quot;</code></pre>
<pre class="r"><code>summary(filt_seurat_object$nCount_iso)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    569    1736    2548    3444    4777   15962 </code></pre>
<pre class="r"><code>print(&quot;gene counts per cell&quot;)</code></pre>
<pre><code>[1] &quot;gene counts per cell&quot;</code></pre>
<pre class="r"><code>summary(filt_seurat_object$nCount_RNA)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1307    3941    5960    8573   12466   38763 </code></pre>
<pre class="r"><code>print(&quot;Number of Cells&quot;) </code></pre>
<pre><code>[1] &quot;Number of Cells&quot;</code></pre>
<pre class="r"><code>dim(filt_seurat_object)[2]</code></pre>
<pre><code>[1] 10268</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Red Hat Enterprise Linux 9.4 (Plow)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS;  LAPACK version 3.10.1

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Matrix_1.7-1                lubridate_1.9.3            
 [3] forcats_1.0.0               dplyr_1.1.4                
 [5] purrr_1.0.2                 readr_2.1.5                
 [7] tidyr_1.3.1                 tibble_3.2.1               
 [9] ggplot2_3.5.1               tidyverse_2.0.0            
[11] patchwork_1.3.0             cowplot_1.1.3              
[13] stringr_1.5.1               DoubletFinder_2.0.4        
[15] BiocParallel_1.40.0         data.table_1.16.2          
[17] gridExtra_2.3               DropletUtils_1.26.0        
[19] SingleCellExperiment_1.28.0 SummarizedExperiment_1.36.0
[21] Biobase_2.66.0              MatrixGenerics_1.18.0      
[23] matrixStats_1.4.1           Seurat_5.1.0               
[25] SeuratObject_5.0.2          sp_2.1-4                   
[27] rtracklayer_1.64.0          GenomicRanges_1.58.0       
[29] GenomeInfoDb_1.42.0         IRanges_2.40.0             
[31] S4Vectors_0.44.0            BiocGenerics_0.52.0        
[33] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22          splines_4.4.0            
  [3] later_1.3.2               BiocIO_1.16.0            
  [5] bitops_1.0-9              R.oo_1.27.0              
  [7] polyclip_1.10-7           XML_3.99-0.17            
  [9] fastDummies_1.7.4         lifecycle_1.0.4          
 [11] edgeR_4.4.0               rprojroot_2.0.4          
 [13] globals_0.16.3            processx_3.8.2           
 [15] lattice_0.22-6            MASS_7.3-61              
 [17] magrittr_2.0.3            limma_3.62.1             
 [19] plotly_4.10.4             sass_0.4.9               
 [21] rmarkdown_2.29            jquerylib_0.1.4          
 [23] yaml_2.3.10               httpuv_1.6.15            
 [25] sctransform_0.4.1         spam_2.10-0              
 [27] spatstat.sparse_3.1-0     reticulate_1.39.0        
 [29] pbapply_1.7-2             RColorBrewer_1.1-3       
 [31] abind_1.4-8               zlibbioc_1.52.0          
 [33] Rtsne_0.17                R.utils_2.12.3           
 [35] RCurl_1.98-1.16           git2r_0.35.0             
 [37] GenomeInfoDbData_1.2.13   ggrepel_0.9.6            
 [39] irlba_2.3.5.1             listenv_0.9.1            
 [41] spatstat.utils_3.1-1      goftest_1.2-3            
 [43] RSpectra_0.16-2           dqrng_0.4.1              
 [45] spatstat.random_3.3-1     fitdistrplus_1.2-1       
 [47] parallelly_1.38.0         DelayedMatrixStats_1.28.0
 [49] leiden_0.4.3.1            codetools_0.2-20         
 [51] DelayedArray_0.32.0       scuttle_1.16.0           
 [53] tidyselect_1.2.1          UCSC.utils_1.2.0         
 [55] farver_2.1.2              spatstat.explore_3.3-1   
 [57] GenomicAlignments_1.40.0  jsonlite_1.8.9           
 [59] progressr_0.15.0          ggridges_0.5.6           
 [61] survival_3.7-0            tools_4.4.0              
 [63] ica_1.0-3                 Rcpp_1.0.13-1            
 [65] glue_1.8.0                SparseArray_1.6.0        
 [67] mgcv_1.9-1                xfun_0.49                
 [69] HDF5Array_1.34.0          withr_3.0.2              
 [71] fastmap_1.2.0             rhdf5filters_1.18.0      
 [73] fansi_1.0.6               callr_3.7.3              
 [75] digest_0.6.37             timechange_0.3.0         
 [77] R6_2.5.1                  mime_0.12                
 [79] colorspace_2.1-1          scattermore_1.2          
 [81] tensor_1.5                spatstat.data_3.1-2      
 [83] R.methodsS3_1.8.2         utf8_1.2.4               
 [85] generics_0.1.3            httr_1.4.7               
 [87] htmlwidgets_1.6.4         S4Arrays_1.6.0           
 [89] whisker_0.4.1             uwot_0.2.2               
 [91] pkgconfig_2.0.3           gtable_0.3.6             
 [93] lmtest_0.9-40             XVector_0.46.0           
 [95] htmltools_0.5.8.1         dotCall64_1.2            
 [97] scales_1.3.0              png_0.1-8                
 [99] spatstat.univar_3.0-0     knitr_1.48               
[101] rstudioapi_0.15.0         tzdb_0.4.0               
[103] reshape2_1.4.4            rjson_0.2.23             
[105] nlme_3.1-166              curl_6.0.0               
[107] rhdf5_2.50.0              cachem_1.1.0             
[109] zoo_1.8-12                KernSmooth_2.23-24       
[111] vipor_0.4.7               parallel_4.4.0           
[113] miniUI_0.1.1.1            ggrastr_1.0.2            
[115] restfulr_0.0.15           pillar_1.9.0             
[117] vctrs_0.6.5               RANN_2.6.2               
[119] promises_1.3.0            beachmat_2.22.0          
[121] xtable_1.8-4              cluster_2.1.6            
[123] beeswarm_0.4.0            evaluate_1.0.1           
[125] locfit_1.5-9.10           cli_3.6.3                
[127] compiler_4.4.0            Rsamtools_2.20.0         
[129] rlang_1.1.4               crayon_1.5.3             
[131] future.apply_1.11.3       labeling_0.4.3           
[133] ps_1.7.5                  ggbeeswarm_0.7.2         
[135] getPass_0.2-4             plyr_1.8.9               
[137] fs_1.6.5                  stringi_1.8.4            
[139] viridisLite_0.4.2         deldir_2.0-4             
[141] munsell_0.5.1             Biostrings_2.72.1        
[143] lazyeval_0.2.2            spatstat.geom_3.3-2      
[145] RcppHNSW_0.6.0            hms_1.1.3                
[147] sparseMatrixStats_1.18.0  future_1.34.0            
[149] Rhdf5lib_1.28.0           statmod_1.5.0            
[151] shiny_1.9.1               highr_0.11               
[153] ROCR_1.0-11               igraph_2.1.1             
[155] bslib_0.8.0              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
